app/controllers/alchemy/admin/pages_controller.rb

```diff
@@ -171,10 +171,18 @@ def unlock
 
       def visit
         @page.unlock!
-        redirect_to show_page_path(
-          urlname: @page.urlname,
-          locale: prefix_locale? ? @page.language_code : nil
-        )
+        if @page.site.host == "*"
+          redirect_to show_page_path(
+            urlname: @page.urlname,
+            locale: prefix_locale? ? @page.language_code : nil,
+          )
+        else
+          redirect_to show_page_url(
+            urlname: @page.urlname,
+            locale: prefix_locale? ? @page.language_code : nil,
+            host: @page.site.host
```

Comment:
AFAIK the `x_url` helper generates urls with the same port as the current request.